File: C/shortest_unsorted_continuous_subarray.c
int compare(const void * a, const void * b){
  return ( *(int*)a - *(int*)b );
}

int* copy_and_sort(int* nums, int numsSize){
    int* copy = calloc(numsSize, sizeof(int));
    for(int i = 0; i < numsSize; i++) *(copy+i) = *(nums+i);
    qsort(copy, numsSize, sizeof(int), compare);
    
    return copy;
}

int findUnsortedSubarray(int* nums, int numsSize){
    if(numsSize == 0) return 0;
        
    int* copy = copy_and_sort(nums, numsSize);
    int right = numsSize-1, left = 0;
    bool isRright = false, isLeft = false;
    for(int i = 0; i < numsSize; i++){
        if(isRright && isLeft) break;
        if(*(copy+i) != *(nums+i) && !isLeft){ 
            left = i;
            isLeft = true;           
        }
        if(*(copy+(numsSize-1-i)) != *(nums+(numsSize-1-i)) && !isRright){ 
            right = numsSize-1-i;
            isRright = true;
        }
    }
    if(!isRright && !isLeft && (right - left == numsSize-1)) return 0;
        
    return right - left + 1;
}

File: Python3/shortest_unsorted_continuous_subarray.py
class Solution:
    def findUnsortedSubarray(self, nums: List[int]) -> int:
        if not nums: return 0
        
        copy = nums.copy()
        copy.sort()
        right, left = len(nums)-1, 0
        isRright, isLeft = False, False
        for i in range(len(nums)):
            if isRright and isLeft: break
            if copy[i] != nums[i] and not isLeft: 
                left = i
                isLeft = True            
            if copy[len(nums)-1-i] != nums[len(nums)-1-i] and not isRright: 
                right = len(nums)-1-i
                isRright = True
        
        if not isRright and not isLeft and (right - left == len(nums)-1): 
            return 0
        
        return right - left + 1
